src/Serilog/Configuration/LoggerDestructuringConfiguration.cs

```diff
@@ -70,6 +71,23 @@ public LoggerConfiguration AsScalar(Type scalarType)
         /// <returns>Configuration object allowing method chaining.</returns>
         public LoggerConfiguration AsScalar<TScalar>() => AsScalar(typeof(TScalar));
 
+        /// <summary>
+        /// When destructuring objects that inherit from Dictionary, serialise them as dictionaries.
+        /// </summary>
+        /// <typeparam name="T">Type to destructure as Dictionary.</typeparam>
+        /// <returns>Configuration object allowing method chaining.</returns>
+        public LoggerConfiguration AsDictionary<T>() where T : IDictionary
+        {
+            TypeInfo typeInfo = typeof(T).GetTypeInfo();
+            Type keyType = typeInfo.GenericTypeArguments[0];
```

Comment:
These will produce `IndexOutOfRangeException` but the failure should be `ArgumentException`

If the type is non-generic (an `IDictionary`) this branch will fail, will need a fallback.